pkg/lib/git/cleanup.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2025 The KitOps Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package git
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+// CleanGitMetadata removes all git-related files from a repository
+func CleanGitMetadata(dir string) error {
+	filesToClean := []string{
+		".git",
+		".gitattributes",
+		".gitignore",
```

Comment:
There can be ".gitignore" file can be on the subdirectories. Should those be handled?

Reply:
This is a good point. Should we scan entire directories in this way? It could become complicated / time-consuming, since we'd have to walk the whole tree just for this, but maybe it's not a huge issue.

I'll add it to the PR.

Reply:
We're now cleaning .gitattributes and .gitignore in any subpath